### PR TITLE
Public docs for Window

### PR DESCRIPTION
**Pull request by Starbuck5**
For #2603 

This moves the docs for Window from sdl2_video.rst to a new window.rst. I've declined to categorize it, so the new module docs shows up in the "other" category on the module index at the top of the documentation.

I also added the following blurb to the Window class description:
> The Window class (formerly known as _sdl2.video.Window), is a newly published feature of pygame-ce 2.4.0. This class allows for programs to drive multiple Windows on-screen at once, something not possible with the [pygame.display.set_mode()](https://github.com/pygame-community/pygame-ce/compare/display.html#pygame.display.set_mode) API. Not everything possible with [pygame.display](https://github.com/pygame-community/pygame-ce/compare/display.html#module-pygame.display) is possible yet in the Window API, but the new window class will continue to be developed, and we're excited to share the new functionality this class offers.

### CODE REVIEW

**Comment by Starbuck5** [comment]
I'm not sure why the linux circle ci is failing, this is a docs PR, the failure must be unrelated.

**Review by MyreMylar — APPROVED** [review]
LGTM 👍

File: src_c/window.c [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include "pgcompat.h"
 
 #include "doc/sdl2_video_doc.h"
```
Comment (ankith26) [inline src_c/window.c:7]: Can this include can be removed now?

Comment (Starbuck5) [reply src_c/window.c:7]: No, because this file still implements an _sdl2.video function `get_grabbed_window`

**Review by ankith26 — COMMENTED** [review]
Generated the docs locally, and while `Window` does show up in the top bar in the base `index.html`, it does not show up when focused on the docs for any other submodule

**Comment by Starbuck5** [comment]
Try running a full regen.

**Review by ankith26 — APPROVED** [review]
LGTM, thanks for the PR :sunglasses:

**Pull request merged**